tests/restapi/conftest.py

```diff
@@ -55,3 +55,92 @@ def restrict_sqlalchemy_queuepool(aiida_profile):
     backend_manager = get_manager().get_backend_manager()
     backend_manager.reset_backend_environment()
     backend_manager.load_backend_environment(aiida_profile, pool_timeout=1, max_overflow=0)
+
+
+@pytest.fixture
+def populate_restapi_database(clear_database_before_test, aiida_localhost):
+    """Populates the database with a considerable set of nodes to test the restAPI"""
+    # pylint: disable=too-many-locals, too-many-statements, unused-argument
+    import tempfile
+    from aiida import orm
+    from aiida.common.links import LinkType
+
+    # create test inputs
+    cell = ((2., 0., 0.), (0., 2., 0.), (0., 0., 2.))
+    structure = orm.StructureData(cell=cell)
+    structure.append_atom(position=(0., 0., 0.), symbols=['Ba'])
+    structure.store()
+    structure.add_comment('This is test comment.')
+    structure.add_comment('Add another comment.')
+
+    cif = orm.CifData(ase=structure.get_ase())
+    cif.store()
+
+    parameter1 = orm.Dict(dict={'a': 1, 'b': 2})
+    parameter1.store()
+
+    parameter2 = orm.Dict(dict={'c': 3, 'd': 4})
+    parameter2.store()
+
+    kpoint = orm.KpointsData()
+    kpoint.set_kpoints_mesh([4, 4, 4])
+    kpoint.store()
+
+    resources = {'num_machines': 1, 'num_mpiprocs_per_machine': 1}
+
+    calcfunc = orm.CalcFunctionNode(computer=aiida_localhost)
+    calcfunc.store()
+
+    calc = orm.CalcJobNode(computer=aiida_localhost)
+    calc.set_option('resources', resources)
+    calc.set_attribute('attr1', 'OK')
+    calc.set_attribute('attr2', 'OK')
+    calc.set_extra('extra1', False)
+    calc.set_extra('extra2', 'extra_info')
+
+    calc.add_incoming(structure, link_type=LinkType.INPUT_CALC, link_label='link_structure')
+    calc.add_incoming(parameter1, link_type=LinkType.INPUT_CALC, link_label='link_parameter')
+
+    aiida_in = 'The input file\nof the CalcJob node'
+    # Add the calcjob_inputs folder with the aiida.in file to the CalcJobNode repository
+    with tempfile.NamedTemporaryFile(mode='w+') as handle:
+        handle.write(aiida_in)
+        handle.flush()
+        handle.seek(0)
+        calc.put_object_from_filelike(handle, 'calcjob_inputs/aiida.in', force=True)
+    calc.store()
```

Comment:
Same here, you are not actually testing any of the files of calcjobnodes in the `test_count_consistency` test, which is the only place you use this fixture don't you?

Reply:
No, you are right. I think originally I thought of using this fixture for all restapi tests; I copied this from the setup in `test_routes.py`, where I put the first tests of this, but then I made the separate `test_statistics.py` for which I only need the nodes. So I can't take all the rest out.